evalml/data_checks/multicollinearity_data_check.py

```diff
@@ -0,0 +1,50 @@
+
+from evalml.data_checks import (
+    DataCheck,
+    DataCheckMessageCode,
+    DataCheckWarning
+)
+from evalml.utils.gen_utils import _convert_to_woodwork_structure
+
+
+class MulticollinearityDataCheck(DataCheck):
+    """Check if any set features are likely to be multicollinear."""
+
+    def __init__(self, threshold=1.0):
+        """Check if any set of features are likely to be multicollinear.
+
+        Arguments:
+            threshold (float): The threshold to be considered. Defaults to 1.0.
+        """
+        if threshold < 0 or threshold > 1:
+            raise ValueError("threshold must be a float between 0 and 1, inclusive.")
+        self.threshold = threshold
+
+    def validate(self, X, y=None):
+        """Check if any set of features are likely to be multicollinear.
+
+        Arguments:
+            X (ww.DataTable, pd.DataFrame, np.ndarray): The input features to check
+
+        Returns:
+            dict: dict with a DataCheckWarning if there are any potentially multicollinear columns.
+
+        """
+        messages = {
+            "warnings": [],
+            "errors": []
+        }
+
+        X = _convert_to_woodwork_structure(X)
+        mutual_info_df = X.mutual_information()
+        if mutual_info_df.empty:
+            return messages
+        above_threshold = mutual_info_df.loc[mutual_info_df['mutual_info'] >= self.threshold]
+        correlated_cols = [(col_1, col_2) for col_1, col_2 in zip(above_threshold['column_1'], above_threshold['column_2'])]
```

Comment:
Will this remove duplicate pairs, i.e. `('col_A', 'col_B')` and `('col_B', 'col_A')` ? Ideally we should find a way to do that.

Reply:
Luckily, `mutual_information` from Woodwork handles this; that's why we only see one pair in the expected test results 😁